packages/core/src/utils/MessageValidator.ts

```diff
@@ -11,4 +14,24 @@ export class MessageValidator {
   public static validate<T extends object>(classInstance: T) {
     return validateOrReject(classInstance)
   }
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  public static validateSync<T extends object>(
```

Comment:
```suggestion
  public static validateSync<T extends Record<string, unknown>>(
```

Reply:
This also won't work when parsing an `AgentMessage`

Reply:
Thats odd. The only reason I can imagine this would not work is when the key is `number` or `Symbol`. Could you try `Record<string | number | Symbol, unknown>`?

If that works we should create a new type alias for that and use that. We use `Record<string, unknown>` everywhere as an `anyObject` type, but it does not catch all objects.

Reply:
I'll give that a try but am suspecting that at least for the .reduce in transformer this will still complain - but lemme check.

Reply:
Other PR - due to complexity